lib/Twig/ExpressionParser.php

```diff
@@ -350,8 +350,18 @@ public function getFunctionNode($name, $line)
                 if (count($args) < 2) {
                     throw new Twig_Error_Syntax('The "attribute" function takes at least two arguments (the variable and the attributes).', $line, $this->parser->getStream()->getSourceContext()->getName());
                 }
+                if (count($args) === 2) {
+                    $arguments = null;
+                } elseif ($args->getNode(2) instanceof Twig_Node_Expression_Array) {
+                    $arguments = new Twig_Node_Expression_Array(array(), $line);
+                    foreach ($args->getNode(2)->getKeyValuePairs() as $pair) {
+                        $arguments->addElement($pair['value']);
+                    }
+                } else {
+                    $arguments = $args->getNode(2);
```

Comment:
this can be a dynamic expression, which means it is still possible to evaluate this node to an associative array